src/microbe_stage/editor/upgrades/ChemoreceptorUpgradeGUI.cs

```diff
@@ -1,124 +1,241 @@
+
+    [Export]
+    public NodePath SpeciesLabelPath = null!;
 
     [Export]
     public NodePath MaximumDistancePath = null!;
 
     [Export]
     public NodePath MinimumAmountPath = null!;
 
+    [Export]
+    public NodePath MinimumAmountLabelPath = null!;
+
     [Export]
     public NodePath ColourPath = null!;
 
 #pragma warning disable CA2213
+    private OptionButton targetTypes = null!;
     private OptionButton compounds = null!;
+    private Label compoundLabel = null!;
+    private OptionButton species = null!;
+    private Label speciesLabel = null!;
     private Slider maximumDistance = null!;
     private Slider minimumAmount = null!;
+    private Label minimumAmountLabel = null!;
     private TweakedColourPicker colour = null!;
 #pragma warning restore CA2213
 
-    private List<Compound>? shownChoices;
+    private List<Compound>? shownCompoundChoices;
+    private List<Species>? shownSpeciesChoices;
+
+    private enum TargetType
+    {
+        Compound,
+        Species,
+    }
 
     public override void _Ready()
     {
+        targetTypes = GetNode<OptionButton>(TargetTypesPath);
         compounds = GetNode<OptionButton>(CompoundsPath);
+        compoundLabel = GetNode<Label>(CompoundsLabelPath);
+        species = GetNode<OptionButton>(SpeciesPath);
+        speciesLabel = GetNode<Label>(SpeciesLabelPath);
         maximumDistance = GetNode<Slider>(MaximumDistancePath);
         minimumAmount = GetNode<Slider>(MinimumAmountPath);
+        minimumAmountLabel = GetNode<Label>(MinimumAmountLabelPath);
         colour = GetNode<TweakedColourPicker>(ColourPath);
 
         compounds.Clear();
+        species.Clear();
 
         maximumDistance.MinValue = Constants.CHEMORECEPTOR_RANGE_MIN;
         maximumDistance.MaxValue = Constants.CHEMORECEPTOR_RANGE_MAX;
-
         minimumAmount.MinValue = Constants.CHEMORECEPTOR_AMOUNT_MIN;
         minimumAmount.MaxValue = Constants.CHEMORECEPTOR_AMOUNT_MAX;
+
+        TypeChanged((int)TargetType.Compound);
     }
 
-    public void OnStartFor(OrganelleTemplate organelle)
+    public void OnStartFor(OrganelleTemplate organelle, GameProperties currentGame)
     {
-        shownChoices = SimulationParameters.Instance.GetCloudCompounds();
+        shownCompoundChoices = SimulationParameters.Instance.GetCloudCompounds();
+        shownCompoundChoices.Sort((x, y) => string.Compare(x.Name, y.Name, StringComparison.OrdinalIgnoreCase));
 
-        foreach (var choice in shownChoices)
+        foreach (var choice in shownCompoundChoices)
         {
             compounds.AddItem(choice.Name);
         }
 
-        // Select glucose by default
-        var defaultCompoundIndex =
-            shownChoices.FindIndex(c => c.InternalName == Constants.CHEMORECEPTOR_DEFAULT_COMPOUND_NAME);
+        shownSpeciesChoices = currentGame.GameWorld.Map.FindAllSpeciesWithPopulation();
+        shownSpeciesChoices.Sort((x, y) => string.Compare(x.FormattedName, y.FormattedName,
```

Comment:
And same here.

Reply:
As [discovered](https://github.com/Revolutionary-Games/Thrive/pull/4261/files#r1313798649), there's probably no better comparison method syntax available here.